webui/react/src/hooks/useSettings.ts

```diff
@@ -1,359 +1,299 @@
+import { Type } from 'io-ts';
 import queryString from 'query-string';
-import { useCallback, useEffect, useMemo, useState } from 'react';
-import { useLocation, useNavigate } from 'react-router-dom';
-
-import usePrevious from 'shared/hooks/usePrevious';
-import {
-  clone,
-  hasObjectKeys,
-  isBoolean,
-  isEqual,
-  isNumber,
-  isObject,
-  isString,
-} from 'shared/utils/data';
-import { Storage } from 'shared/utils/storage';
+import { useCallback, useContext, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { useStore } from 'contexts/Store';
+import { updateUserSetting } from 'services/api';
+import { UpdateUserSettingParams } from 'services/types';
+import { Primitive } from 'shared/types';
+import { isEqual } from 'shared/utils/data';
+import { ErrorType } from 'shared/utils/error';
 import handleError from 'utils/error';
 
-import { Primitive, RecordKey, ValueOf } from '../shared/types';
-import { ErrorType } from '../shared/utils/error';
-
-import useStorage from './useStorage';
-
-export const BaseType = {
-  Boolean: 'Boolean',
-  Float: 'Float',
-  Integer: 'Integer',
-  Object: 'Object',
-  String: 'String',
-} as const;
-
-export type BaseType = ValueOf<typeof BaseType>;
-
-const PathChangeType = {
-  None: 'none',
-  Push: 'push',
-  Replace: 'replace',
-} as const;
-
-type PathChangeType = ValueOf<typeof PathChangeType>;
-
-type GenericSettingsType = Primitive | Primitive[] | Record<number, number[]> | undefined;
-type GenericSettings = Record<string, GenericSettingsType>;
-type PathChange<T> = { querySettings: Partial<T>; type: PathChangeType };
-
-/*
- * defaultValue     - Optional default value. `undefined` as ultimate default.
- * skipUrlEncoding  - Avoid preserving setting in the URL query param.
- * storageKey       - If provided, save/load setting into/from storage.
- * type.baseType    - How to decode the string-based query param.
- * type.isArray     - List based query params can be non-array.
- */
-export interface SettingsConfigProp {
-  defaultValue?: GenericSettingsType;
-  key: string;
+import { Settings, SettingsProvider, UserSettings } from './useSettingsProvider';
+
+export enum BaseType {
```

Comment:
we should use object literal instead of enum.

Reply:
Should actually be removed since we're using io-ts.